rust/foxglove/src/websocket_server.rs

```diff
@@ -77,6 +83,19 @@ impl WebSocketServer {
         self
     }
 
+    /// Configure TLS with a PEM-formatted x509 certificate chain and pkcs8 private key.
+    /// If enabled, the server will only accept connections using wss://.
+    /// If TLS configuration fails, starting the server will result in an error.
+    #[doc(hidden)]
+    #[cfg(feature = "tls")]
+    pub fn tls(mut self, cert: &[u8], key: &[u8]) -> Self {
```

Comment:
How do you feel about making this directly take a TlsIdentity struct? It would prevent accidentally mixing up the order of the arguments.